osu.Game.Rulesets.Osu.Tests/TestCaseSliderInput.cs

```diff
@@ -0,0 +1,287 @@
+// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System.Collections.Generic;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Beatmaps;
+using osu.Game.Beatmaps.ControlPoints;
+using osu.Game.Replays;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Replays;
+using osu.Game.Rulesets.Replays;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Screens.Play;
+using osu.Game.Tests.Beatmaps;
+
+namespace osu.Game.Rulesets.Osu.Tests
+{
+    public class TestCaseSliderInput : SliderTestBase
+    {
+        private readonly Container content;
+        protected override Container<Drawable> Content => content;
+
+        protected override List<Mod> Mods { get; set; }
+
+        private Player player;
+        private JudgementResult lastResult;
+        private bool allJudgedFired;
+
+        public TestCaseSliderInput()
+        {
+            Mods = new List<Mod>();
+            base.Content.Add(content = new OsuInputManager(new RulesetInfo { ID = 0 }));
+        }
+
+        /// <summary>
+        ///     Pressing a key before a slider, pressing the other key on the slider head, then releasing the latter pressed key
+        ///     should result in tracking to end.
+        ///     At 250ms intervals:
+        ///     Frame 1 (prior to slider):          Left Click
+        ///     Frame 2 (within slider hit window): Left & Right Click
+        ///     Frame 3 (while tracking):           Left Click
+        ///     A passing test case will have the cursor lose tracking on frame 3.
+        /// </summary>
+        [Test]
+        public void TestLeftBeforeSliderThenRight()
+        {
+            AddStep("Invalid key transfer test", () =>
+            {
+                var actions = new List<List<OsuAction>>();
```

Comment:
please check my previous review, all these have not been converted to object initialisers.